tools/sr-snapshots/Sources/Shell/ShellCommand.swift

```diff
@@ -0,0 +1,80 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+public struct ShellError: Error, CustomStringConvertible {
+    let result: ShellResult
+
+    public var description: String {
+        return """
+        status: \(result.status)
+        output: \(result.output ?? "")
+        error: \(result.error ?? "")
+        """
+    }
+}
+
+/// Result of executing shell command.
+public struct ShellResult {
+    /// Command's STDOUT value.
+    public let output: String?
+    /// Command's STDERR value.
+    public let error: String?
+    /// Exit code of the command.
+    public let status: Int32
+}
+
+/// Executes given shell command and returns STDOUT.
+/// Throws if command ends with status code other than `0`.
+@discardableResult
+public func shell(_ command: String) throws -> String {
+    let result = try shellResult(command)
+    if result.status != 0 {
+        throw ShellError(result: result)
+    } else if let output = result.output, !output.isEmpty {
+        return output
+    } else if let error = result.error, !error.isEmpty {
+        return error
+    } else {
+        return ""
+    }
+}
+
+/// Executes given shell command.
+@discardableResult
+public func shellResult(_ command: String) throws -> ShellResult {
```

Comment:
optional

you can reduce the string processing by observing the notification. For reference https://github.com/DataDog/apple-symbols-harvester/blob/ganeshnj/feat/RUM-3299-firmware-extraction/Sources/HarvesterFirmware/CommandLine.swift

Reply:
One benefit you get is async printing of the output. In current version, the output is printed at the end of execution which is a great experience when longer operations are in play.

Reply:
I like it, will do 👍

Reply:
Done ✅ - I used your implementation @ganeshnj as it was more versatile than mine attempt.